### PR TITLE
Batch update: copy max_* params to resourcelimit directive

### PR DESCRIPTION
**Pull request by jfy133**
This is in preparation for the new nf-core/tools 3.0.0 release, that deprecates the `check_max()` function + `max_()` params with the native Nextflow 'resourceLimit' directive.

I have done an initial pass using a bash script, but there are some remaining profiles that need to be addressed.

This will need to be dealt with manually.

| Done | Config                       | Issue                   |
|---------|------------------------------|-------------------------|
|   x        |conf/alice.config            | Multiple process blocks |
|     x       |conf/arcc.config             | Multiple process blocks |
|     x       |conf/azurebatch.config       | Multiple process blocks |
|      x      |conf/azurebatchdev.config    | Multiple process blocks |
|   x         |conf/crg.config              | Multiple process blocks |
|     x       |conf/czbiohub_aws.config     | Multiple process blocks |
|      x      |conf/eva.config              | Multiple process blocks |
| x          | conf/hasta.config            | Multiple process blocks |
|    x       | conf/hki.config              | Multiple process blocks |
|    x        |conf/imperial.config         | Multiple process blocks |
|    x       | conf/incliva.config          | Multiple process blocks |
|   x        | conf/maestro.config          | Multiple process blocks |
|  x         | conf/medair.config           | Multiple process blocks |
|   x         |conf/pawsey_nimbus.config    | Multiple process blocks |
|   x         |conf/rosalind_uge.config     | Multiple process blocks |
|   x        | conf/sage.config             | Multiple process blocks |
|     x       | conf/sanger.config           | Multiple process blocks |
|   x        | conf/uge.config              | Multiple process blocks |
|     x     | conf/uppmax.config           | Multiple process blocks |
|     x      | conf/vsc_calcua.config       | Multiple process blocks |
|     x       |conf/vsc_kul_uhasselt.config | Multiple process blocks |
|    x       | conf/vsc_ugent.config        | Multiple process blocks |
|       x    | crukmi                       | Uses check_max()        |
|     x      | rosalind_uge                     | Unused check_max()      |
|       x   | sage                         | Uses check_max()        |
|      x      |sahmri                       | Uses check_max()        |
|     x      | uge                          | Unused check_max()      |
|        x   | utd_ganymede                 | Uses check_max()        |

To be adressed in follow up PR

| Done | Config                       | Issue                   |
|---------|------------------------------|-------------------------|
|           | pipeline/eager/eva           | Uses check_max()        |
|           | pipeline/eager/maestro       | Uses check_max()        |
|           | pipeline/eager/mag           | Uses check_max()        |
|           | pipeline/rnafusion/hasta     | Uses check_max()        |
|           | pipeline/taxprofiler/hasta   | Uses check_max()        |

Scripts and log:
[batch-resourcelimit-scripts.zip](https://github.com/user-attachments/files/17103524/batch-resourcelimit-scripts.zip)

### CODE REVIEW

**Comment by jfy133** [comment]
@BrunoGrandePhD @edmundmiller @nathanhaigh Would it be OK for your clusters if I remove the `check_max()` function calls in your respective institutional level pipelines? 

This function is being removed from nf-core active and future pipelines very soon, and thus will break your config. 

This PR adds the new 'native' Nextflow way of doing this - so I would remove the `check_max()` and replace with the new `resourceLimits` equivalent.

If it's absolutely required, then we can copy the function into your config - but allowing custom functions will be removed very soon.

File: conf/crukmi.config [hunk 1]
```diff
@@ -26,23 +26,23 @@ process {
     maxRetries     = 3
 
     withLabel: process_single {
-        cpus   = 1
-        memory = 5
+        cpus   = { 1 * task.attempt }
+        memory = { 5 * task.attempt }
```
Comment (SPPearce) [inline conf/crukmi.config:30]: Hmm, this one doesn't have `.GB`?
Shall we make it consistent:
```suggestion
        memory = { 5.GB * task.attempt }
```

**Comment by edmundmiller** [comment]
Go for it for my configs!

File: conf/uppmax.config [hunk 2]
```diff
@@ -84,12 +89,23 @@ if (hostname.matches("^s[0-9][0-9]*")) {
     params.max_time = 700.h
     params.max_memory = 3880.GB
     params.config_profile_description = 'UPPMAX (Snowy) cluster profile provided by nf-core/configs.'
+    process {
+            resourceLimits = [
+                memory: 3880.GB,
```
Comment (mahesh-panchal) [inline conf/uppmax.config:94]:
```suggestion
                memory: 3880.GB,
                cpus: 16,
```

File: conf/uppmax.config [hunk 3]
```diff
@@ -84,12 +89,23 @@ if (hostname.matches("^s[0-9][0-9]*")) {
...
+    }
 }
 
 // Cluster: Irma
 if (hostname.startsWith("i")) {
     params.max_memory = 250.GB
     params.config_profile_description = 'UPPMAX (Irma) cluster profile provided by nf-core/configs.'
+    process {
+        resourceLimits = [
+            memory: 250.GB,
```
Comment (mahesh-panchal) [inline conf/uppmax.config:106]:
```suggestion
            memory: 250.GB,
            cpus: 16,
            time: 240.h
```

File: conf/uppmax.config [hunk 4]
```diff
@@ -98,13 +114,26 @@ if (hostname.startsWith("m")) {
...
 
 // Cluster: Rackham
 if (hostname.startsWith("r")) {
     params.max_cpus = 20
     params.max_memory = 970.GB
     params.config_profile_description = 'UPPMAX (Rackham) cluster profile provided by nf-core/configs.'
+    process {
+        resourceLimits = [
+            memory: 970.GB,
+            cpus: 20
```
Comment (mahesh-panchal) [inline conf/uppmax.config:134]:
```suggestion
            cpus: 20,
            time: 240.h
```

File: conf/uppmax.config [hunk 5]
```diff
@@ -120,6 +149,12 @@ profiles {
             max_time = 1.h
         }
         executor.queueSize = 1
+        process {
+            resourceLimits = [
+                memory: 120.GB,
```
Comment (mahesh-panchal) [inline conf/uppmax.config:154]:
```suggestion
                memory: 120.GB,
                cpus: 4,
```

**Review by mahesh-panchal — COMMENTED** [review]
I think the uppmax profile should have the full resource limits map set. I have a feeling inheritance doesn't apply here.

**Comment by jfy133** [comment]
@HaidYi I've had to tweak your St. Jude profile slightly to make the tests pass (not sure why it didn't come up in your original PR), it bascially uses a Groovy call to resolve the TMPDIR path, because otherwise Nextflow thinks it's it's own variable.

I hope that's OK

**Review by pontus — APPROVED** [review]

**Review by JoseEspinosa — APPROVED** [review]
LGTM! 🚀

**Review by edmundmiller — APPROVED** [review]

**Review by mahesh-panchal — APPROVED** [review]

**Pull request merged**